### PR TITLE
Don't reroute exceptions thrown in expectAsync

### PR DESCRIPTION
**Pull request by natebosch**
Allows using `expectAsync*` to wrap callbacks in context where they are
called with a `try/catch` without changing visible behavior.

### CODE REVIEW

**Comment by natebosch** [comment]
I'm not 100% sure of the full impact of this or whether we should consider it breaking...

**Comment by natebosch** [comment]
This is covered by a test which looks like it was in place from the beginning:

https://github.com/dart-lang/test/commit/0a098e7f895842db421b7fad29eb1de0d1047ee8

There doesn't appear to be any specific discussion about this in [the review](https://codereview.chromium.org//934413002) and it's a large CL...

The `catch` and `return null` go back all the way to at least https://github.com/dart-lang/test/commit/c55a076b1d642185e5456a9792685d805133c058

Reviewed in https://codereview.chromium.org//869043002

This was a refactoring from https://codereview.chromium.org/869043002/diff/20001/lib/src/spread_args_helper.dart which did not have a `return null`.

It's hard to know for sure, but I _think_ the behavior might be a holdover from when this code served some other purpose and isn't strictly needed...

**Comment by kevmoo** [comment]
Reasonable to throw it through Tap internally?

**Comment by grouma** [comment]
Already ran a TAP presubmit.

**Comment by kevmoo** [comment]
Sweet!

*From: *Gary Roumanis <notifications@github.com>
*Date: *Wed, May 8, 2019 at 1:46 PM
*To: *dart-lang/test
*Cc: *Kevin Moore, Comment

Already ran a TAP presubmit.

File: pkgs/test_api/test/frontend/expect_async_test.dart [hunk 1]
```diff
@@ -300,30 +300,12 @@ void main() {
     });
   });
 
-  group("with errors", () {
-    test("reports them to the current test", () async {
```
Comment (jakemac53) [inline pkgs/test_api/test/frontend/expect_async_test.dart]: It seems like we should restore this test (checking that async errors are reported)?

Comment (natebosch) [reply pkgs/test_api/test/frontend/expect_async_test.dart]: I don't think we want them to be reported. It should be allowed to have some callback somewhere throw and not fail the test.

Comment (natebosch) [reply pkgs/test_api/test/frontend/expect_async_test.dart]: Ah, I may have misunderstood.

I don't think that this test was useful for testing the behavior of async errors. That should be covered elsewhere.

Comment (jakemac53) [reply pkgs/test_api/test/frontend/expect_async_test.dart]: Ok, I don't have enough context here to know it just seems wrong on the surface to delete that test :man_shrugging:

**Pull request merged**